src/agent/library/skills.js

```diff
@@ -1340,3 +1340,39 @@ export async function activateNearestBlock(bot, type) {
     log(bot, `Activated ${type} at x:${block.position.x.toFixed(1)}, y:${block.position.y.toFixed(1)}, z:${block.position.z.toFixed(1)}.`);
     return true;
 }
+
+
+export async function digDown(bot, distance = 10) {
+    /**
+     * Digs down a specified distance.
+     * @param {MinecraftBot} bot, reference to the minecraft bot.
+     * @param {int} distance, distance to dig down.
+     * @returns {Promise<boolean>} true if successfully dug down.
+     * @example
+     * await skills.digDown(bot, 10);
+     **/
+
+    for (let i = 0; i < distance; i++) {
+        const targetBlock = bot.blockAt(bot.entity.position.offset(0, -1, 0));
+        const belowBlock = bot.blockAt(bot.entity.position.offset(0, -2, 0));
+
+        // Check for lava, water, or a fall of more than 5 blocks below the bot
+        if (!targetBlock || targetBlock.name === 'lava' || targetBlock.name === 'water' || 
+            (belowBlock && (belowBlock.name === 'lava' || belowBlock.name === 'water' || belowBlock.position.y < bot.entity.position.y - 5))) {
```

Comment:
belowBlock will only ever be -2 below the bot. you will need to search down through air blocks until you reach a solid block or too many air blocks to determine if there is danger of falling.